### ISSUE TITLE
Most controls aren't accessible to blind users via screen readers like NVDA

### ISSUE THREAD

**Issue by amirsol81** [r0x0r/pywebview#545]
### Specification
- pywebview version: 
3.2
- platform / version: 
Windows 10 Pro 64-Bit, version 2004
### Description  
  Dear developers,

Having used the latest r0x0r/pywebview, I noticed that with screen readers for the visually impaired like NVDA, navigation via arrow keys is virtually impossible. That is, with NVDA enabled, pressing Arrow keys doesn't move the focus, and one can navigate around only via Tab and SHIFT+Tab. Even these key strokes don't land on many controls. Equally problematic is the fact that once one returns to r0x0r/pywebview, via ALT+Tab, NVDA faces difficulty giving it the proper focus, meaning even Tab and SHIFT+Tab fail to work as expected. To take care of these, I suggest the following for proper screen reader access:
1. Pywebview should allow screen readers to navigate via Up, Down, Left and Right arrows -- like what is possible with any HTML/Chromium environment. Full Arrow-key access is of utmost importance in order to see and interact with essential sections and areas of any app which uses  Pywebview.
2. Tab and SHIFT+Tab should land on many more controls.
3. Returning focus to the environment via ALT+Tab or ALT+SHIFT+Tab should properly give screen readers the focus needed to interact with all environment controls.

Thanks for your efforts.
Cheers
  
  
### Practicalities

- YES/NO I am willing to work on this issue myself.
I'm not a developer.
- YES/NO I am prepared to support this issue financially.
Sadly I can't do it myself.

**Comment by r0x0r** [comment]
What is NVDA? What web renderer do you use? 
1. I don't know if this can be done.
2. Like what controls? Currently it lands on links and buttons. Does extra `tabindex` work for you?
3. The problem seems to be with EdgeHTML, with MSHTML it works as it should. Have you tested CEF?

**Comment by RemixDev** [comment]
NVDA or (NonVisual Desktop Access) is a screen reader software for windows, used for making the desktop interface usable for blind users.It's an app that only works on Windows, so only the windows renderer are affected.

In my app I'm using the cef renderer and it's working fine on external browsers, but on pywebview I had reports of users not being able to use correctly the app with their screen readers (like the alt tab issue).

I think the controls he means are the Aria tags (some extra tags that are needed for screen reader to read the screen correctly). I still need to investigate it.

**Comment by amirsol81** [comment]
As a visually impaired NVDA User on Windows, I think the most problematic issue with pywebview is that we can't locate links and blocks of text as we can't navigate the environment built around pywebview with Up/Down arrow keys. As such, we can't interact with those important blocks or simply read them either.
And my experience indicates only some buttons are focused via Tab and SHIFT+Tab, and links are also ignored.
To get an idea, I suggest that you download NVDA from https://www.nvaccess.org/files/nvda/releases/2020.1/nvda_2020.1.exe , install it, and run it to see how it interacts with HTML-oriented apps like Google Chrome, Firefox, and even Internet Explorer. It's a light-weight app -- less than 24MB. Basically NVDA's behavior in pywebview should be similar to, say, Chromium apps but it isn't at the moment.

**Comment by r0x0r** [comment]
Have you tried CEF renderer for pywebview if it is any different?

>

**Comment by amirsol81** [comment]
This app on Windows is using the CEF renderer, resulting in the accessibility issues described above.

**Comment by SaschaCowley** [comment]
This is an issue with cefpython (or possibly cef), not pywebview.

**Comment by amirsol81** [comment]
So if this is a cefpython or cef issue, to whom or where it should be reported? It is a serious issue affecting visually impaired users.

**Comment by SaschaCowley** [comment]
Yes, it is. And it should be reported against cef or cefpython, depending where the issue is.

**Comment by okevinalmeida** [comment]
Hi! After creating a duplicate and being pointed to this issue, I did some tests and the problem isn't on pywebview, but on the renderer it uses by default. To make it accessible here I needed to do: webview.start(gui='mshtml'). That did the trick for me here, but needs confirmation

**Comment by SaschaCowley** [comment]
MSHTML or EdgeHTML are (I believe) both accessible, and are used by default on Windows. Note that there may be problems with keyboard shortcuts with MSHTML (and I believe also EdgeHTML) (#325 )

**Comment by amirsol81** [comment]
@okevinalmeida Thanks for your awesome efforts! So is this going to
find its way into the next release of pywebview, or should developers
do something else to implement/incorporate it into their projects? I'm
more of an accessibility tester and that's why I'm asking it.

**Comment by r0x0r** [comment]
@amirsol81 I would not hold my breathe this is getting fixed anytime soon, unless somebody else provides a fix.

The current realities of this project are
1. I am the sole developer. 
2. I do not get paid for this.
3. It is summer time.
4. I am not very interested in this issue.

So essentially there are two ways to give your issue a boost. Either do it yourself or sponsor it.

**Comment by r0x0r** [comment]
I looked into the details and it seems to be doable to enable accessibility controls in CEF. The work would take 2 hours at 70€/h + VAT, making $228.

The open collective expense can be found here  https://opencollective.com/pywebview/expenses/24063 

You can sponsor this issue by contributing the pywebview collective.
